<?xml version="1.0" encoding="UTF-8"?>
<bm:model xmlns:e="aurora.service.exception" xmlns:o="aurora.database.local.oracle" xmlns:bm="http://www.aurora-framework.org/schema/bm" xmlns:f="aurora.database.features" alias="t1" baseTable="VAT_SELLER_COMPANIES">
    <bm:fields>
        <bm:field name="id" databaseType="NUMBER" datatype="java.lang.Long" physicalName="ID" prompt="VAT_SELLER_COMPANIES.ID"/>
        <bm:field name="seller_id" databaseType="NUMBER" datatype="java.lang.Long" physicalName="SELLER_ID" prompt="VAT_SELLER_COMPANIES.SELLER_ID"/>
        <bm:field name="company_id" databaseType="NUMBER" datatype="java.lang.Long" physicalName="COMPANY_ID" prompt="VAT_SELLER_COMPANIES.COMPANY_ID"/>
        <bm:field name="company_code" expression="(select company_code from fnd_companies_vl v1 where v1.company_id = t1.company_id)" forInsert="false" forQuery="true" forUpdate="false" prompt="机构代码"/>
        <bm:field name="company_name" expression="(select company_short_name from fnd_companies_vl v2 where v2.company_id = t1.company_id)" forInsert="false" forQuery="true" forUpdate="false" prompt="机构名称"/>
    </bm:fields>
    <bm:features>
        <f:standard-who/>
        <o:sequence-pk sequenceName="vat_seller_companies_s"/>
    </bm:features>
    <bm:primary-key>
        <bm:pk-field name="id"/>
    </bm:primary-key>
    <e:exception-descriptor-config>
        <e:exception-descriptor exception="java.sql.SQLException" handleClass="aurora.database.SQLExceptionDescriptor">
            <e:error-message code="1" message="机构重复!"/>
        </e:exception-descriptor>
    </e:exception-descriptor-config>
    <bm:query-fields>
        <bm:query-field name="seller_id" queryExpression="t1.seller_id = ${@seller_id} "/>
        <bm:query-field name="company_code_from" queryExpression="company_code &gt;= ${@company_code_from} "/>
        <bm:query-field name="company_code_to" queryExpression="company_code &lt;= ${@company_code_to}"/>
    </bm:query-fields>
</bm:model>
